include/circt/Dialect/Moore/MooreOps.td

```diff
@@ -246,11 +249,23 @@ def NetOp : MooreOp<"net", [
   );
   let results = (outs Res<RefType, "", [MemAlloc]>:$result);
   let assemblyFormat = [{
-    ``custom<ImplicitSSAName>($name) $kind ($assignment^)? attr-dict
+    `` custom<ImplicitSSAName>($name) $kind ($assignment^)? attr-dict
     `:` type($result)
   }];
 }
 
+def AssignedVarOp : MooreOp<"assigned_variable", [
```

Comment:
Do we need a new op? NetOp seems to have an `assignment` operand so is that same?

Reply:
You're totally right! `NetOp` has an `assignment` operand. Maybe I think I need to rename it. Like `def AssignedDeclOp : MooreOp<"assigned_decl(aration)", [`, what do @fabianschuiki  @uenoku  think? By the way, slang will throw an error, if we drive the same variable multiple times.

Reply:
Hmmm yeah good question... SV differentiates between nets and variables in quite a few places. My gut feeling is that turning a `VariableOp` into a `NetOp` with an assignment isn't going to be correct in general :thinking:. The `AssignedVariableOp` feels like a good step, because it retains the variable-ness but makes it explicit that it is continuously assigned a specific value.

I'm pretty sure we'll be able to write some simplification passes that take `always_comb` procedures that write to variables through `moore.blocking_assign` and convert that into a simple `moore.assigned_variable` for simplicity. Then only the truly annoying uses of variables would remain (the ones interacting with the event queue, have delays, etc.).